### PR TITLE
torch_nntile: implement repeat via chained scale_slice broadcast

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Adds a `scale_slice`-only broadcast layer in `torch_nntile` and wires it to PyTorch's `aten::repeat` for `device="nntile"`.

- **`broadcast_scale_slice_chain`**: expands a lower-rank tensor to a target shape by chaining one `scale_slice` per destination dimension (same pattern as cross-entropy backward grad broadcast).
- **`repeat_scale_slice_chain`**: implements `Tensor.repeat()` by inserting one broadcast axis at a time with `scale_slice` (handles PyTorch left-padding of repeat dims and virtual leading dimensions).
- **`aten::repeat`**: registered on `PrivateUse1` so `.repeat()` records into `TensorGraph` and runs through StarPU.

## Testing

```bash
export LD_LIBRARY_PATH=$PWD/build/nntile:/opt/starpu/lib
pytest -vv torch_nntile/tests/test_repeat_parity.py
```

Covers 1D/2D repeat, padded repeat dimensions, identity `repeat(1, 1)`, and leading-dimension padding (`[N].repeat(1, 2, 1)`).

## Notes

- Float32 contiguous tensors only (consistent with other phase-2 torch_nntile ops).
- Identity repeat (`repeat(1, 1, …)`) falls back to `memcpy` when no `scale_slice` ops are needed.

<div><a href="https://cursor.com/agents/bc-992011ae-1446-46be-915a-e2bc2a47b0e7"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-992011ae-1446-46be-915a-e2bc2a47b0e7"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review
